include/executor.h

```diff
@@ -43,11 +44,16 @@ namespace detail {
 
 	  public:
 		// TODO: Try to decouple this more.
-		executor(
-		    node_id local_nid, host_queue& h_queue, device_queue& d_queue, task_manager& tm, buffer_manager& buffer_mngr, reduction_manager& reduction_mngr);
+		executor(const size_t num_nodes, const node_id local_nid, host_queue& h_queue, device_queue& d_queue, task_manager& tm, buffer_manager& buffer_mngr,
+		    reduction_manager& reduction_mngr);
 
 		void startup();
 
+		void enqueue(command_pkg&& pkg) {
+			std::scoped_lock lk(m_command_queue_mutex);
```

Comment:
That's the first time ever I've seen a `std::scoped_lock` being used, but sure, why not! :grinning:

Reply:
Look. There's too many damn lock types in the standard library 😛